pkg/columns/formatter/textcolumns/scaler.go

```diff
@@ -58,46 +62,88 @@ func (tf *TextColumnsFormatter[T]) RecalculateWidths(maxWidth int, force bool) {
 	tf.currentMaxWidth = maxWidth
 
 	// Get total width of all printed columns
-	totalWidth := 0
-	spaces := 0
+	totalWidthNotFixed := 0
+	totalWidthFixed := 0
 	for i, column := range tf.showColumns {
 		if i > 0 {
-			spaces += len([]rune(tf.options.ColumnDivider))
+			totalWidthFixed += len([]rune(tf.options.ColumnDivider))
 		}
```

Comment:
Maybe you can initialize `totalWidthFixed := (len(tf.showColumns) - 1) * len([]rune(tf.options.ColumnDivider))` to avoid having this conditional on the loop?